tripal/tripal
# Fixed loadOntology function in the GFF3 importer to not return if it …

# Bug Fix

Issue #926 

## Description



As reported by @mpoelchau, the GFF importer would not import all of the Ontology terms if one of them did not already exist in the database.  The importer would just stop trying once it found a problem.  The example is if there are InterPro terms in a GFF3 file mixed with GO terms.  If GO is loaded by the IPR terms are not then the first IPR term encountered would cause import of ontology terms to stop.    For example:

```
QKKF02000028.1	Genbank	mRNA	3267	4896	.	+	.	locus_tag=LSTR_LSTR016833;ID=LSTR016833-RA;Parent=LSTR016833;Ontology_term=KEGG:K05096,InterPro:IPR002290,InterPro:IPR020635,GO:0004672,GO:0004713,InterPro:IPR001245,GO:0006468,InterPro:IPR000719,GO:0005524;
```

The simple fix was to not return from the load ontology function and just continue trying to process the other terms in the list.  I did make a few other changes that are very minor:  1) I just adjusted some commenting that was incorrectly formated.  2) I just happen to notice that some of the update functions weren't wrapped in a try catch statement, so I added that.

## Testing?




1) On a test Tripal site, create a new test organism and analysis.
2) Make sure you have the Gene Ontology loaded.
3) Import the test.fasta sequence below.  (I couldn't upload a file with a .fasta extension so I had to add an extra .txt on the end.  You'll want to remove that.)
4) Import the test.gff file below
5) Publish the `mRNA` content type.
6) Check the mRNA page to make sure that all of the GO terms are loaded.  The IPR terms should be missing.  You should have these terms: GO:0004672, GO:0004713, GO:0006468, GO:0005524

Here are test files:


[test.fasta.txt](https://github.com/tripal/tripal/files/3117565/test.fasta.txt)
[test.gff.txt](https://github.com/tripal/tripal/files/3117566/test.gff.txt)


## Screenshots (if appropriate):
![PRsc](https://user-images.githubusercontent.com/1719352/56746623-6cec6180-6731-11e9-96da-4f6196ddd96e.png)

## Additional Notes (if any):
